apps/web/src/components/BridgeForm.tsx

```diff
@@ -130,28 +130,48 @@ export default function BridgeForm({
   const { getBalance } = useCheckBalance();
   const [isBalanceSufficient, setIsBalanceSufficient] = useState(true);
   const [tokenBalances, setTokenBalances] = useState({});
+  const [isVerifyingTransaction, setIsVerifyingTransaction] = useState(false);
 
-  const checkBalance = debounce(async () => {
+  async function getBalanceFn() {
     const key = `${selectedNetworkA.name}-${selectedTokensA.tokenB.symbol}`;
     const balance = await getBalance(selectedTokensA.tokenB.symbol);
-    setTokenBalances({
+    const updatedBalances = {
       ...tokenBalances,
       [key]: balance,
-    });
+    };
+
+    setTokenBalances(updatedBalances);
+    return updatedBalances;
+  }
+
+  const checkBalance = debounce(async () => {
+    getBalanceFn();
   }, 200);
 
-  useEffect(() => {
+  async function verifyTransfer(refetch?: boolean) {
```

Comment:
```suggestion
  async function verifySufficientHWBalance(
    refetch?: boolean
  ): Promise<boolean | undefined> {
```
Is this func name better? verifyTransfer sounds a bit generic